site/e2e/helpers.ts

```diff
@@ -490,6 +489,16 @@ const createTemplateVersionTar = async (
             token: randomUUID(),
             ...agent,
           } as Agent;
+
+          try {
+            Agent.encode(agentResource);
+          } catch (e) {
+            throw new Error(
+              `agentResource encode failed, missing defaults?\n${e}\n${e.stack}`,
```

Comment:
Oh, wait – forgot there's one other issue. Since JS/TS allow you to throw any kind of value at all, TS doesn't assume that a thrown value is actually an error, and just defaults to type `unknown`

You'd need a type guard to prove to the compiler that the value is an error before you try accessing error-specific properties from it
```ts
try {
  // Throw-able code
} catch (e) {
  let errorMessage  = `agentResource encode failed, missing defaults?\n${e}`;
  if (e instanceof Error) {
    errorMessage += `\n${e.stack}`;
  }
   
  throw new Error(errorMessage);
}
```

Reply:
Obviously, because JS, the `${e.name}: ${e.message}` is sometimes included in the stack trace (Node, etc.) and other times not (browser). Hopefully we now have a satisfying solution with the last commit. 😄